backend/src/main/java/com/llm_service/llm_service/service/ConversationService.java

```diff
@@ -35,26 +34,21 @@ public Optional<Conversation> getByID(UUID id) {
     }
 
     // TODO optimize this fetching mechanism
-    public List<Discussion> askLLMQuestion(UUID id, ConversationRequest conversationRequest)
-            throws ConversationNotFoundException {
-        Optional<Conversation> conversation = conversationPersistenceManager.findById(id);
-
-        if (conversation.isEmpty()) {
-            throw new ConversationNotFoundException(id);
-        }
+    public List<Discussion> askLLMQuestion(ConversationRequest conversationRequest, Conversation conversation) {
```

Comment:
maybe we should rename this `ConversationRequest` -> `LLMQuestionRequest` that way the function does not have two parameters that seems the same but are not.